iptables/table.go

```diff
@@ -560,11 +596,60 @@ func (t *Table) getHashesFromBuffer(buf *bytes.Buffer) map[string][]string {
 	return newHashes
 }
 
-func (t *Table) InvalidateDataplaneCache() {
+func (t *Table) InvalidateDataplaneCache(reason string) {
+	log.WithField("reason", reason).Info("Invalidating dataplane cache")
 	t.inSyncWithDataPlane = false
 }
 
-func (t *Table) Apply() {
+var (
+	postUpdateRefreshDelays = []time.Duration{
+		1000 * time.Millisecond,
+		500 * time.Millisecond,
+		200 * time.Millisecond,
+		100 * time.Millisecond,
+		50 * time.Millisecond,
+	}
+)
+
+func (t *Table) Apply() (rescheduleAfter time.Duration) {
+	if t.inSyncWithDataPlane {
+		// We _think_ we're in sync, check if there are any reasons to think we might
+		// not be in sync.
```

Comment:
Yes, I wasn't too happy with it but also struggling to see a great way to simplify; I'll noodle it some more